### PR TITLE
Clear Include before `updateRecord`

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: Web/Controller/LandingPages.hs [hunk 1]
```diff
@@ -79,6 +72,11 @@ instance Controller LandingPagesController where
 
                     setSuccessMessage "LandingPage updated"
                     redirectTo EditLandingPageAction { .. }
+                    where
+                        extractLandingPageFromInclude :: Include' ["paragraphCtas", "paragraphQuotes"] LandingPage -> LandingPage
+                        extractLandingPageFromInclude landingPage = landingPage
```
Comment (amitaibu) [inline Web/Controller/LandingPages.hs:77]: image

Comment (mpscholten) [reply Web/Controller/LandingPages.hs:77]: can you try to split this into two sub functions: one turning `Include' ["paragraphCtas", "paragraphQuotes"] LandingPage -> Include' ["paragraphCtas"] LandingPage` and then one turning `Include' ["paragraphCtas"] LandingPage -> LandingPage` ?

Comment (amitaibu) [reply Web/Controller/LandingPages.hs]: That actually works, which isn't intuitive. https://github.com/amitaibu/ihp-landing-page/pull/3/commits/a9f132ae2ea68765bc24a3437de8ecb1f628717d

Why is that (and how did you figure it might work :smile: )?

Is it somehow related to the fact that the compiler thinks that `Include "paragraphQuotes" LandingPage` is not equal to `Include' ["paragraphQuotes"] LandingPage`

Comment (mpscholten) [reply Web/Controller/LandingPages.hs]: I expected this by intuition 😄 In very early IHP versions `updateField` was actually `set` and it was later changed (because it was causing errors like these). The core problem with `updateField` is, that it's type definition is very open. It's defined as [`updateField :: value' -> model -> model'`](https://github.com/digitallyinduced/ihp/blob/master/IHP/HaskellSupport.hs#LL246C5-L246C45). Here `model` and `model'` are two independent type variables. This means a call to `updateField` can actually change the return type of the record (e.g. turning `LaindingPage` to `Include "paragraphQuotes" LandingPage`). The problem with your first version was that GHC could not figure out the `model'` variable because multiple `updateField` were chained (the `model` type argument is easy for GHC to figure out, it's just that the `model'` is independent of that).

For comparison `set` is defined as `set :: value -> model -> model`. In that case when GHC can figure out `model` it will never error. So `set` is typically easy to figure out and unlikely to error. One the other side the problem with `set` is that it's less flexible and cannot change the output type (e.g. with `model = LandingPage` it will be `set :: value -> LandingPage -> LandingPage`, so there's no way to express e.g. the `Include` stuff). That's why `updateField` exists.

TL;DR: `updateField` has a more open type signature than `set` and sometimes GHC needs a bit of help

Comment (amitaibu) [reply Web/Controller/LandingPages.hs]: :ok_hand: Thanks for the explanation.

Comment (amitaibu) [reply Web/Controller/LandingPages.hs]: @mpscholten I've updated Stackoverflow, so this info is not lost :)

https://stackoverflow.com/a/76241756/750039

Comment (mpscholten) [reply Web/Controller/LandingPages.hs]: Thanks 👍
